libraries/core_libs/network/rpc/eth/Eth.cpp

```diff
@@ -276,18 +273,19 @@ class EthImpl : public Eth, EthParams {
     throw std::runtime_error(result.consensus_err.empty() ? result.code_err : result.consensus_err);
   }
 
-  void set_transaction_defaults(TransactionSkeleton& t, EthBlockNumber blk_n) {
+  // this should be used only in eth_call and eth_estimateGas
+  void prepare_transaction_for_call(TransactionSkeleton& t, EthBlockNumber blk_n) {
     if (!t.from) {
       t.from = ZeroAddress;
     }
     if (!t.nonce) {
       t.nonce = transaction_count(blk_n, t.from);
     }
     if (!t.gas_price) {
-      t.gas_price = gas_pricer();
+      t.gas_price = 0;
     }
     if (!t.gas) {
-      t.gas = 90000;
+      t.gas = gas_limit;
```

Comment:
lets use dag block gas limit as we talked - it is pretty easy and quick fix to pass this config param to Eth class